### ISSUE TITLE
Private repository cloudrepo.io 401 Unauthorized

### ISSUE THREAD

**Issue by imatw4r** [astral-sh/uv#8510]
Hey folks! I am running to a problem installing dependencies with private registry hosted on [cloudrepo](https://www.cloudrepo.io/).
The command I execute is:
```
uv pip install -r requirements.txt --extra-index-url https://<USERNAME>:<PASSWORD>@[redacted].mycloudrepo.io/repositories/[redacted]
```
That causes:
```
Failed to download `[redacted-package-name]==0.0.13`
  Caused by: HTTP status client error (401 Unauthorized) for url (https://[redacted].mycloudrepo.io/repositories/[redacted]/[redacted-package-name]/[redacted-package-name]-0.0.13-py2.py3-none-any.whl)
```

Executing pip command:
```
pip install -r requirements.txt
```
With ENV variable `PIP_EXTRA_INDEX_URL=` set to the same value I pass via `--extra-index-url` to UV works, and the dependency is found and installed.

Note: I pass password & username directly as a string, not trying to interpolate (I am aware of #5734 discussion). I also tried setting an environment variable UV_PIP_EXTRA_INDEX_URL but it doesn't work.

I know there was an issue reported (#1709) that seems to be closed already, but I wonder if it was supposed to be fixed for all platforms (azure/aws, etc) or if it's needed to be added separately for `mycloudrepo.io`?

Some more details:

Platform:
MacOS, Sonoma 14.4
M1 Apple Silicon

```
uv --version
> uv 0.4.22 (Homebrew 2024-10-15)
```

**Comment by charliermarsh** [comment]
The environment variable you're looking for is `UV_EXTRA_INDEX_URL` (no `_PIP_`), though passing on the command-line should be working too. It's hard to say more without being able to debug the index response itself...

**Comment by zanieb** [comment]
Can you share redacted logs with `RUST_LOG=uv=trace`? That'll give us more information about the authentication flow.

**Comment by imatw4r** [comment]
@zanieb thanks for the info about `RUST_LOG=trace`. It helped me solved the issue.

TLDR in my `requirements.txt` I had specified:

```
# Add our private package repo, with read only credentials.
--extra-index-url https://${CLOUDREPO_USERNAME}:${CLOUDREPO_PASSWORD}@[REDACTED].mycloudrepo.io/repositories/goodrx

package1==version
package2==version
```

And seems like index specified in `requirements.txt` has higher priority over the one passed explicitly via `--extra-index-url` when calling `uv pip install`, which caused an issue for me as I didn't have these env variables specified.

It works now, sorry for the inconvenience @zanieb @charliermarsh
